src/components/organisms/Badges/Badges.tsx

```diff
@@ -22,14 +22,17 @@ import { WithId } from "utils/id";
 import { venueInsideUrl, venuePreviewUrl } from "utils/url";
 import { notEmpty } from "utils/types";
 
+import { BadgeImage } from './BadgeImage'
+
 export const Badges: React.FC<{
   user: WithId<User>;
   currentVenue: WithId<Venue>;
 }> = ({ user, currentVenue }) => {
   const [visits, setVisits] = useState<WithId<UserVisit>[]>([]);
   const [venues, setVenues] = useState<WithId<AnyVenue>[]>([]);
+  const [loading, setLoading] = useState<boolean>(false);
 
-  const imgRef: RefObject<HTMLImageElement[]> = useRef([]);
+  const badgeImgRefs: RefObject<HTMLImageElement[]> = useRef([]);
```

Comment:
Is this only used inside of the `BadgeImage` component now? If so, might make sense to put it in there as well? I think this would also resolve my concerns about seemingly mutating an array passed as a prop.